packages/ui/core/src/UiWalletHandle.ts

```diff
@@ -0,0 +1,6 @@
+import type { IdentifierArray } from '@wallet-standard/base';
+
+export type UiWalletHandle = {
+    readonly '~uiWalletHandle': unique symbol;
+    readonly features: IdentifierArray;
```

Comment:
`features` is only an array on `WalletAccount` and an object on `Wallet`, right? I would think `features` should be included in the `Pick` for `UiWalletAccount` and `UiWallet` separately, but I assume I'm misunderstanding something here

Reply:
I see the intention is to have them go through `useWalletFeature` or similar to get a reference to the feature from the handle. I'm not sure why this is necessary (aside from the fact that the actual functions can't be serialized, is that what's important for something here?) but it does seem confusing that the type of the `features` property in `UiWallet` doesn't match `Wallet`.

Reply:
I can explain, officer!

So many of the wallet feature functions demand that you bring an account as an argument.

```ts
features: {
    'standard:signMessage': {
        async signMessage(config: { account: WalletAccount, message: Uint8Array }) {
            /* ... */
        },
    }
}
```

Event though `signMessage` is literally a feature on the wallet, it's _conceptually_ a feature of the account. You don't sign messages with wallets, you sign them with wallet accounts.

So I'm a component in an app, and all you've passed me is an account.

```tsx
const SignMessagePanel({ account }) {
    return <button onClick={() => {
        /* oh crap, I need `wallet.features` here */
    }}>Sign it</button>;
}
```

Damnit, you're missing the wallet, where the actual feature is located.

One way we could do this is:

```tsx
const SignMessagePanel({ account, wallet }) {
    return <button onClick={() => {
        wallet.features.signMessage({
            account,
            message: new TextEncoder().encode('Hello world'),
        }).then(console.log);
    }}>Sign it</button>;
}
```

This is bad for a bunch of reasons:

* Now you're _demanding_ that the app pass around the `(wallet, account)` tuple and never lose track of either, which is extra work
* We've opened the possibility of the _wrong wallet_ being passed here, meaning you'd pass an account from Wallet A into a `feature` on Wallet B. Bad.

OK why don't we just bind everything together on `UiWalletAccount` and make both `UiWallet` and `UiWalletAccount` have an object of features?

```tsx
function createUiWalletAccount(account: WalletAccount, wallet: Wallet) {
    return {
        features: {
            'standard:signMessage': {
                async signMessage(args: { message: Uint8Array }) {
                    return await wallet.features.signMessage({ ...args, account });
                }
            },
        },
    };
}
```

This is bad for even more reasons:

* I can't construct that shim function unless I enumerate every possible feature in existence (ie. I have to know where the `account` needs to go in the target feature to be able to curry it there)
* Lots of code in the bundle people might never use
* `Wallet` is now hard-linked to the `WalletAccount` (can't be garbage collected, memory leaks, etc)

So the tack I decided to take was: _nobody_ gets to cart their features around with them; just a flat array of feature names.

* Apps can still feature detect on wallets _and_ accounts (eg. filter out all wallet accounts that don't support `standard:encrypt`)
* Apps don't need to bookkeep the walllet/account pair, which eliminates work and a source of bugs
* It's literally impossible now to screw up by, say, choosing an account that doesn't belong to a wallet, because you have two wallets connected, both having an account with the same address
* Passing around a `UiWalletAccount` is sufficient for nearly everything an app has to do (aside from connecting and disconnecting, which is not an account-level operation)

Reply:
Makes sense. I think it could confuse a few people because in all other ways a `Wallet` quacks like a `UiWallet`, but it's probably fine since they aren't really expected to touch the Standard directly now through these abstractions, so the fact that they are different will be lost on most.